### ISSUE TITLE
[Feature Request] Gym spec conversion modifying key names

### ISSUE THREAD

**Issue by matteobettini** [pytorch/rl#1418]
In the gym to torchrl spec transform https://github.com/pytorch/rl/blob/main/torchrl/envs/libs/gym.py#L181
the key "state" of dict spaces is overwritten with a new name. 

Is it necessary to do it in this location? users might want to transform their gym specs to torchrl without this side effect.

**Comment by vmoens** [comment]
Yep it isn't perfect. We can change that but it'll be bc breaking so I would be happy with any solution that does not break things before the 0.3 release (leaving users with 0.2 with a warning)

**Comment by matteobettini** [comment]
One simple one is adding an argument like parse_state_key with default true?
